.github/workflows/docker-build-and-deploy.yml

```
@@ -0,0 +1,138 @@
+# Copyright (C) 2021 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+
+name: Docker Build and Deploy
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+
+jobs:
+  docker_push:
+    name: Build Docker Image
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout main repository
+        uses: actions/checkout@v3
+
+      - name: Add version to gha env
+        run: cat .versions >> $GITHUB_ENV
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Identify ORT revision build
+        run: |
+          echo "Building image with revision ${GITHUB_SHA::10}"
+          echo "ORT_VERSION=${GITHUB_SHA::10}" >> $GITHUB_ENV
+
+      - name: Set organization
+        run: echo "ORG_BASE_NAME=${GITHUB_REPOSITORY_OWNER}" >> $GITHUB_ENV
+
+      # Ort basse
+      - name: Extract components metadata (tags, labels) for base image
+        id: meta_base
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.REGISTRY }}/${{ env.ORG_BASE_NAME }}/base
+
+      - name: Identify base image
+        run: echo "Will be tagged as ${{ steps.meta.outputs.tags }} and labeled as ${{ steps.meta.outputs.labels }}"
+
+      - name: Build ort base container
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          target: base
+          load: true #${{ github.event_name != 'pull_request' }}
```

```suggestion
          load: ${{ github.event_name != 'pull_request' }}
```
Reminder to change it before merging :wink: